### ISSUE TITLE
Clarify Admin Role usage

### ISSUE THREAD

**Issue by chpapa** [SkygearIO/features#131]
# Description

In Skygear, there is a special role called `admin`. In [Skygear docs](https://docs.skygear.io/guides/cloud-db/acl-overview/js/), it was said to be:

> What only admin role can do:
> Assign role to user
> Revoke role to user

However, Admin role seems also able to override ACL as it was used in Skygear CMS.

I think we should clearly define the permission of admin role and shall break it down. From what I understand now, it seems there is at least several permission:

* assign roles, from roles to users
* revoke roles, from roles to users
* access all public DB
* access all private DB
* change all user passwords

In short term, we should at least clarify what is an Admin Role, so users don't accidentally open unnecessary access to all users. Maybe have a few sub-roles to allow finer ACL.

A 2.0 idea is simply switch to policy based ACL for these...

@rickmak please chips in if I misunderstood anything. The above was based on a discussion with @limouren on CMS permissions.
